core/Symbols.cc

```diff
@@ -536,6 +544,14 @@ bool Symbol::addMixin(const GlobalState &gs, ClassOrModuleRef sym) {
     return isValidMixin;
 }
 
+u2 Symbol::addMixinPlaceholder(const GlobalState &gs) {
+    ENFORCE(isClassOrModule());
+    ENFORCE(ref(gs) != core::Symbols::StubMixin(), "Created a cycle through StubMixin");
+    mixins_.emplace_back(core::Symbols::StubMixin());
```

Comment:
I'm re-using this existing sentinel value but if it's important we could create a new one. I tried using an empty value but that causes problems in checks that expect all `mixins_` to exist like `derivesFrom`.

Reply:
Yeah i'd kind of like another symbol here. Right now, if you print the symbol table at some point and see `StubMixin` in the output, you know where sorbet went wrong. Re-using `StubMixin` here makes it trickier to debug in the future.

Also, if we changed this to something like `MixinPlaceholder` here, we could add an ENFORCE later down the line that says that there are no `MixinPlaceholder`'s in the mixins list, which forces Sorbet to have to fill them back in, or replace them with `StubMixin` if need be (This could potentially be in the same place that you mentioned was enforce'ing from the first attempt when using `Symbols::noSymbol()`).

Reply:
I'll look into it. I'm pretty sure we could do that enforce when globalpass linearizes the classes.